src/Servers/Kestrel/shared/test/StreamExtensions.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO
+{
+    public static class StreamExtensions
+    {
+        /// <summary>
+        /// Fill the buffer until the end of the stream.
+        /// </summary>
+        public static async Task<int> FillBufferUntilEndAsync(this Stream stream, byte[] buffer, CancellationToken cancellationToken = default)
+        {
+            var offset = 0;
+            int read;
+
+            do
+            {
+                read = await stream.ReadAsync(buffer, offset, buffer.Length - offset, cancellationToken);
+                offset += read;
+            } while (read != 0 && offset < buffer.Length);
+
+            if (read != 0)
+            {
+                Assert.Equal(0, await stream.ReadAsync(new byte[1], 0, 1, cancellationToken));
+            }
+
+            return offset;
+        }
+
+        public static async Task<int> FillEntireBufferAsync(this Stream stream, byte[] buffer, CancellationToken cancellationToken = default)
```

Comment:
Does anything call this now?

Reply:
![image](https://user-images.githubusercontent.com/303201/128097968-93728ecd-abe2-4fbe-a720-b557d9e18f00.png)

Reply:
[Yep](https://github.com/dotnet/aspnetcore/blob/jamesnk/streams-readmethods/src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs#L1728).